src/test/groovy/com/athaydes/spockframework/report/internal/HtmlReportCreatorSpec.groovy

```diff
@@ -193,10 +201,18 @@ class HtmlReportCreatorSpec extends ReportSpec {
         new File( this.class.classLoader.getResource( cssPath ).toURI() ).text
     }
 
-    private String expectedHtmlReport() {
-        def rawHtml = HtmlReportCreator.getResource( 'FakeTestReport.html' ).text
-        def binding = [
-                classOnTest     : FakeTest.class.name,
+    private String expectedHtmlReport( Class specification ) {
+        def rawHtml = HtmlReportCreator.getResource( "${specification.simpleName}Report.html" ).text
+        def binding = fakeTestBinding( specification )
+        if( specification == VividFakeTest ){
```

Comment:
It's not clear when we use this method that this method knows about each specific case, so this will easily become confusing later when we don't remember that... can you maybe add the bindings to the examples? Or create a Map where you look up bindings for each example... just don't use something like this because it's hidden in a private method and I don't want to have to look at each private method to understand what the test will do.